File: String Matching/Q3. Repeated String Match/repeatedStringMatch.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int repeatedStringMatch(string a, string b) {
        string str = a;
        int cnt = 1;

        while(str.size() < b.size()){
            str += a;
            cnt+=1;
        }
        if(str.contains(b))return cnt;
        str+=a;
        return str.contains(b) ? cnt+1 : -1;
    }
};

File: String Matching/Q3. Repeated String Match/repeatedStringMatch.js
/**
 * @param {string} a
 * @param {string} b
 * @return {number}
 */
var repeatedStringMatch = function (a, b) {
    let str = a
    let cnt = 1;

    while (str.length < b.length) {
        str += a
        cnt++
    }

    if (str.includes(b)) return cnt;
    str += a
    return str.includes(b) ? cnt + 1 : -1
};